graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Functions converting TensorBay Catalog to Graviti Schema."""
+
+from functools import reduce
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterator, List, Union
+
+import yaml
+from tensorbay.dataset import Notes, RemoteData
+from tensorbay.label import (
+    Catalog,
+    ClassificationSubcatalog,
+    InstanceMaskSubcatalog,
+    Keypoints2DSubcatalog,
+    Label,
+    LabeledKeypoints2D,
+    PanopticMaskSubcatalog,
+)
+from tensorbay.label.basic import SubcatalogBase, _LabelBase
+from tensorbay.label.supports import AttributesMixin, CategoriesMixin, MaskCategoriesMixin
+
+_FILE_TYPES_AND_SUFFIX = {
+    ("image", "file.Image"): {".bmp", ".jpg", ".jpeg", ".png", ".gif"},
+    ("point_cloud", "file.PointCloudBin"): {".bin"},
+    ("point_cloud", "file.PointCloud"): {".pcd", ".ply", ".las", ".laz"},
+    ("audio", "file.Audio"): {".mp3", ".wav", ".wma", ".aac"},
+}
+
+_JSON_TYPE_TO_PORTEX_TYPE = {
+    "array": "array",
+    "boolean": "boolean",
+    "integer": "int32",
+    "number": "float32",
+    "string": "string",
+}
+
+_LABEL_PROCESSORS: Dict[str, Callable[..., Any]] = {}
+
+
+def _get_category_params(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    category_params: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        category_params["categoryIds"] = subcatalog.get_index_to_category()
+    else:
+        category_params["categories"] = list(subcatalog.categories.keys())
+
+    if getattr(subcatalog, "category_delimiter", None):
+        category_params["categoryDelimiter"] = subcatalog.category_delimiter
+
+    return category_params
+
+
+def _get_attribute_params(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attribute_params: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        field: Dict[str, Any] = {"name": attribute.name}
+        if hasattr(attribute, "enum"):
+            field["type"] = "enum"
+            field["values"] = attribute.enum
```

Comment:
if attribute type is `enum`, seems there is no need to check minimum and maximum?